setup.py

```diff
@@ -114,7 +114,18 @@ def run(self):
 if any(a.startswith(('bdist', 'build', 'install')) for a in sys.argv):
     from ipykernel.kernelspec import write_kernel_spec, make_ipkernel_cmd, KERNEL_NAME
 
-    argv = make_ipkernel_cmd(executable='python')
+    # If the IPYKERNEL_USE_SYS_EXECUTABLE environment variable is set to 1
+    # then `make_ipkernel_cmd` will generate a kernelspec including the full
+    # sys.executable path.
+    #
+    # Otherwise, a generic 'python' is used instead.
+
+    if os.getenv('IPYKERNEL_USE_SYS_EXECUTABLE'):
```

Comment:
Why an env var and not a proper option? This will be very hard to find out for other projects unaware of this change.

Also, why not making this the default?

Reply:
Regarding the default, the only issue is that (AFAIK), python wheels don't do any relocation.

So having the kernel spec in the wheel for ipykernel merely "python" seems to be the way to go.

Reply:
Rather than a boolean, maybe `IPYTHON_KERNELSPEC_PYTHON` and use the value in the env? e.g. `IPYTHON_KERNELSPEC_PYTHON=$PYTHON python setup.py install`

This seems more complicated and harder to find than the `python -m ipykernel install` command that always works.

If we do this, it would be worth adding a comment here about why we have to use `python` as the default in order to make wheels portable, and that the flag is really only for use with conda-build.
Ind

Reply:
I agree that it is harder to find, but the old way should still work in the same fashion.

I merely want to enable that option for the conda recipe as per the issue mentioned above so that the conda package does not require `jupyter-kernel.exe` on windows.

Reply:
> Yeah, although the issue is that it does not work for some reason with the latest conda.

That's a much more significant issue, though, and shouldn't be relevant to this package: an executable from a different package (jupyter-client) is not being installed in a form that works at all. Changing this behavior to get past a fundamentally broken dependency package is probably not the right fix, in my view.